File: LeetCode/src/array/rotateImage/RotateImage.cpp
#include <vector>

using namespace std;

void rotate(vector<vector<int>>& matrix) {
	int n = matrix.size();
	auto temp = matrix;
	for (int j = n - 1; j >= 0; --j) {
		for (int i = 0; i < n; ++i) {
			temp[i][j] = matrix[n - 1 - j][i];
		}
	}
	matrix = temp;
}

File: LeetCode/src/array/rotateImage/RotateImage.java
package array.rotateImage;

public class RotateImage {
	public void rotate(int[][] matrix) {
		if (matrix == null || matrix.length == 0)
			return;
		int n = matrix.length;
		int[][] temp = new int[n][n];
		for (int i = 0; i < n; i++)
			for (int j = 0; j < n; j++)
				temp[i][j] = matrix[i][j];
		for (int j = n - 1; j >= 0; --j) {
			for (int i = 0; i < n; ++i) {
				matrix[i][j] = temp[n - 1 - j][i];
			}
		}
	}
}
